### PR TITLE
Added concurrency unittest (for creation) - related #4504

### PR DESCRIPTION
**Pull request by noctarius**

### CODE REVIEW

**Comment by serkan-ozal** [comment]
@noctarius can you sync this PR with the current state of master branch ?

File: hazelcast-client/src/test/java/com/hazelcast/client/cache/ClientJCacheConcurrencyTest.java [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+                        e.printStackTrace();
+                    } finally {
+                        endLatch.countDown();
+                    }
+                }
+            }).start();
+        }
+
+        startSemaphore.release(100);
+        endLatch.await(10, TimeUnit.MINUTES);
```
Comment (pveentjer) [inline hazelcast-client/src/test/java/com/hazelcast/client/cache/ClientJCacheConcurrencyTest.java]: We have an assertOpenEventually. If you need to wait potentially 10 minutes for this test to complete, this test has is now allowed to be a normal unit test but needs to be moved to the nightly tests.

File: hazelcast/src/test/java/com/hazelcast/cache/JCacheConcurrencyTest.java [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+                        e.printStackTrace();
+                    } finally {
+                        endLatch.countDown();
+                    }
+                }
+            }).start();
+        }
+
+        startSemaphore.release(100);
+        endLatch.await(10, TimeUnit.MINUTES);
```
Comment (pveentjer) [inline hazelcast/src/test/java/com/hazelcast/cache/JCacheConcurrencyTest.java]: Same here. If you need to wait so long, this test can't be a QuickTest.

File: hazelcast/src/test/java/com/hazelcast/cache/JCacheConcurrencyTest.java [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+                    try {
+                        // Sync all threads
+                        startSemaphore.acquire();
+
+                        // Create CacheManagers
+                        CacheManager cacheManager = cachingProvider1.getCacheManager(uri, null, properties);
+                        managers.set(index, cacheManager);
+
+                    } catch (Exception e) {
+                        e.printStackTrace();
```
Comment (pveentjer) [inline hazelcast/src/test/java/com/hazelcast/cache/JCacheConcurrencyTest.java]: Is this desirable? If you have an exception, your test won't fail. If you are expecting a certain exception, you need to catch the specific exception and name the variable as ignore/expected.

File: hazelcast/src/test/java/com/hazelcast/cache/JCacheConcurrencyTest.java [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+                    try {
+                        // Sync all threads
+                        startSemaphore.acquire();
+
+                        // Create Statistics
+                        CacheStatistics statistics = cache.getLocalCacheStatistics();
+                        stats.set(index, statistics);
+
+                    } catch (Exception e) {
+                        e.printStackTrace();
```
Comment (pveentjer) [inline hazelcast/src/test/java/com/hazelcast/cache/JCacheConcurrencyTest.java]: Same here.

File: hazelcast/src/test/java/com/hazelcast/cache/JCacheConcurrencyTest.java [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+                    } finally {
+                        endLatch.countDown();
+                    }
+                }
+            }).start();
+        }
+
+        startSemaphore.release(100);
+        endLatch.await(10, TimeUnit.MINUTES);
```
Comment (pveentjer) [inline hazelcast/src/test/java/com/hazelcast/cache/JCacheConcurrencyTest.java]: Make use of assertOpenEventually

**Comment by noctarius** [comment]
Guys I'm not in engineering anymore and this stuff is open for a long long long long time, can someone please take this over? I just don't have time to rewrite it another 3 times. Otherwise I will just abandon it now.

**Comment by pveentjer** [comment]
I'm going to close this one. The review comments have not been addressed + rebase is required. Feel free to reopen it.

**Pull request closed without merging**